src/test/scala/akka/persistence/jdbc/query/CurrentEventsByPersistenceIdTest.scala

```diff
@@ -133,6 +137,43 @@ abstract class CurrentEventsByPersistenceIdTest(config: String) extends QueryTes
       }
     }
   }
+
+  it should "allow updating events (for data migrations)" in withActorSystem { implicit system =>
+    val journalOps = new JavaDslJdbcReadJournalOperations(system)
+    val journal = Persistence(system).journalFor("")
+
+    withTestActors() { (actor1, _, _) =>
+      actor1 ! 1
+      actor1 ! 2
+      actor1 ! 3
+
+      eventually {
+        journalOps.countJournal.futureValue shouldBe 3
+      }
+
+      val pid = "my-1"
+      journalOps.withCurrentEventsByPersistenceId()(pid, 1, 3) { tp =>
+        tp.request(Int.MaxValue)
+          .expectNext(EventEnvelope(Sequence(1), pid, 1, 1))
+          .expectNext(EventEnvelope(Sequence(2), pid, 2, 2))
+          .expectNext(EventEnvelope(Sequence(3), pid, 3, 3))
+          .expectComplete()
+      }
+
+      // perform in-place update
+      val journalP = TestProbe()
+      journal.tell(JdbcAsyncWriteJournal.InPlaceUpdateEvent(pid, 1, Integer.valueOf(111)), journalP.ref)
```

Comment:
is the `Integer` because of `AnyRef` ?
wouldn't `111` be boxed anyway?

Reply:
Yeah had some issue there around that, will check again